pkg/build/registry/buildlog/rest.go

```diff
@@ -77,18 +77,27 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
 			return &genericrest.LocationStreamer{}, nil
 		}
 		glog.V(4).Infof("Build %s/%s is in %s state, waiting for Build to start", build.Namespace, name, build.Status.Phase)
-		err := r.waitForBuild(ctx, build)
+		latest, ok, err := registry.WaitForRunningBuild(r.Watcher, ctx, build, r.Timeout)
 		if err != nil {
-			return nil, err
+			switch latest.Status.Phase {
+			case api.BuildPhaseError:
+				return nil, errors.NewBadRequest(fmt.Sprintf("build %s encountered an error: %s", build.Name, buildutil.NoBuildLogsMessage))
```

Comment:
These errors should be the same as the errors returned if the build errored or was cancelled directly (outter switch cases).